src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,50 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints()
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    def _parse_entrypoint(self, entrypoint: str) -> Dict[str, str]:
+        """
+        Extract the entry-point and name from the string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
+        if "=" not in entrypoint:
+            raise ValueError("Entry point %r is not of the form 'name = target'" % entrypoint)
+
+        (name, target) = entrypoint.split("=", 1)
+        return {name.strip(): target.strip()}
+
+    def _generate_entrypoints(self) -> Dict[str, Dict[str, str]]:
+        """
+        Dump the entry-points for the project of the given type, if any.
```

Comment:
"given type" seems a bit out of place since this method doesn't accept any parameters. A minimal change might be something like this:

```suggestion
        Dump the entry-points for the project, if any, organized by type.
```

although I personally think it's kind of understood that they will be organized by type so you could just omit that.

```suggestion
        Dump the entry-points for the project, if any.
```

Then again, if you make any of the other changes I'm suggesting below, a different way of describing it might wind up being more appropriate.

Reply:
Yeah, originally it was going to take a parameter, the entry point type (e.g. `console_scripts`, `gui_scripts` or arbitrary module names), and was going to return the modules _of that type_, then I realised I'd have to make another function to return what entry point types are available.

So dropping the parameter and making it iterate over the lot was cleaner, but forgot to clean up the documentation for it.

Making the other method a static method, this too could be simplified a little bit to make unit testing easier.